core-bundle/src/Resources/contao/library/Contao/Controller.php

```diff
@@ -1180,6 +1180,8 @@ protected static function replaceOldBePaths($strContext)
 
 		$generate = static function ($route) use ($router)
 		{
+			trigger_deprecation('contao/core-bundle', '4.0', 'Using old backend paths has been deprecated in Contao 4.0 and will be removed in Contao 5. Use the backend routes instead.');
+
```

Comment:
This will always trigger the deprecation even if no old backend path was used.

Reply:
You sould use the router service directly so you shouldn't even land here.

Reply:
But the core itself calls the `replaceOldBePaths()` method. IMO we have to check if there is actually an old backend path present before triggering the deprecation...

Reply:
I would think this only triggers if the regex callback is executed for a match?

Reply:
There is no regex or regex callback anywhere in this method? I’m confused now 🙈

Reply:
I think @aschempp is referring to the anonymous function. However, as we are using `str_replace()`, the anonymous function is triggered for every array member. https://3v4l.org/8jZIK

So @ausi is right here.

Reply:
Then the whole method is very stupid because we are generating all routes for no reason. They should only be generated if the key is found, so we must replace this with `preg_replace_callback`. Imho that's a patch worth for Contao 4.9

Reply:
I’d prefer not using words like “stupid”, especially in public code reviews.

Reply:
see https://github.com/contao/contao/pull/4091 for a fix that will make this deprecation usable 🙃